bufferevent_sock.c

```diff
@@ -498,6 +498,41 @@ bufferevent_connect_getaddrinfo_cb(int result, struct evutil_addrinfo *ai,
 }
 
 int
+bufferevent_socket_set_src_address(struct bufferevent *bev,
+    const struct sockaddr *addr, int socklen)
+{
+	struct bufferevent_private *bufev_p =
+	    EVUTIL_UPCAST(bev, struct bufferevent_private, bev);
+
+	evutil_socket_t fd;
+	int r = 0;
+	int result = -1;
+
+	bufferevent_incref_and_lock_(bev);
+
+	if (!bufev_p || !addr)
+		goto done;
+
+	/* do we need to check to see if the socket is connecting/connected
+	 * already?
+	 */
+
+	fd = bufferevent_getfd(bev);
+	if (fd < 0)
+		goto done;
+
+	r = evutil_socket_bind_address(fd, addr, socklen);
+	if (r < 0)
+		goto done;
+
+	result = 0;
+
+done:
+	bufferevent_decref_and_unlock_(bev);
+	return result;
+}
```

Comment:
> Can you convert `evhttp` to use this? (i.e. `evhttp_connection_connect_()`)

Not sure I follow.  `evhttp_connection_connect_()` just calls `bind_socket()`... which in turn calls `bind_socket_ai()`.  The latest set of changes has `bind_socket_ai()` calling `evutil_socket_bind()`.

Is that adequate?